src/Lumper.Lib/BSP/IO/BspFileReader.cs

```diff
@@ -99,8 +99,8 @@ protected override void ReadHeader()
     // Finding the real gamelump length by looking at the next lump
     private void UpdateGameLumpLength()
     {
-        Lump gameLump = null;
-        LumpHeader gameLumpHeader = null;
+        Lump? gameLump = null;
+        LumpHeader? gameLumpHeader = null;
         foreach ((Lump? lump, LumpHeader? header) in Lumps.OrderBy(x => x.Item2.Offset))
```

Comment:
Save/load stuff changed all this behaviour, null handling ended up a lot better. Like discussed I probably shouldn't just dropped all the null handling changes for code not touched by future commits, sorry. Since you've gotten to the newer stuff already I'm gonna leave this

Reply:
Not initializing is a compile error!